crates/autopilot/src/protocol/fee.rs

```diff
@@ -0,0 +1,37 @@
+/// Protocol fee implementation.
+use {
+    crate::{
+        arguments,
+        driver_model::solve::{fee_policy_to_dto, FeePolicy},
+    },
+    model::{
+        auction::Auction,
+        order::{OrderClass, OrderUid},
+    },
+    std::collections::HashMap,
+};
+
+/// Prepares the fee policies for each order in the auction.
+/// Determines if the protocol fee should be applied to the order.
+pub fn fee_policies(
+    auction: &Auction,
+    config: arguments::FeePolicy,
+) -> HashMap<OrderUid, Vec<FeePolicy>> {
+    auction
+        .orders
+        .iter()
+        .map(|order| {
```

Comment:
This could be a `.filter_map()` so we don't unnecessarily create entries in the map for orders that don't need any fees.

Reply:
Indeed.